packages/dev/inspector-v2/src/components/properties/particles/particleSystemProperties.tsx

```diff
@@ -1,63 +1,1063 @@
+import { SphereParticleEmitter } from "core/Particles/EmitterTypes/sphereParticleEmitter";
+import { useCallback, useEffect, useMemo, useState } from "react";
 
 import { useInterceptObservable } from "../../../hooks/instrumentationHooks";
+import { useProperty } from "../../../hooks/compoundPropertyHooks";
 import { useObservableState } from "../../../hooks/observableHooks";
-import { Color4GradientList, FactorGradientList } from "shared-ui-components/fluent/hoc/gradientList";
+import { Color3GradientList, Color4GradientList, FactorGradientList } from "shared-ui-components/fluent/hoc/gradientList";
+import { FileUploadLine } from "shared-ui-components/fluent/hoc/fileUploadLine";
 import { AttractorList } from "./attractorList";
 import { MessageBar } from "shared-ui-components/fluent/primitives/messageBar";
+import { StringifiedPropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/stringifiedPropertyLine";
+import { ButtonLine } from "shared-ui-components/fluent/hoc/buttonLine";
+import { BoundProperty } from "../boundProperty";
+import { BlendModeOptions, ParticleBillboardModeOptions } from "shared-ui-components/constToOptionsMaps";
+import { NumberDropdownPropertyLine, StringDropdownPropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/dropdownPropertyLine";
+import { NumberInputPropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/inputPropertyLine";
+import { Color4PropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/colorPropertyLine";
+import { SwitchPropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/switchPropertyLine";
+import { TextPropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/textPropertyLine";
+import { Vector3PropertyLine } from "shared-ui-components/fluent/hoc/propertyLines/vectorPropertyLine";
+import { LinkToEntityPropertyLine } from "../linkToEntityPropertyLine";
+import { Tools } from "core/Misc/tools";
+import { ParticleSystem as ParticleSystemClass } from "core/Particles/particleSystem";
+import { ParticleHelper } from "core/Particles/particleHelper";
 
-export const ParticleSystemEmissionProperties: FunctionComponent<{ particleSystem: ParticleSystem }> = (props) => {
+const SnippetDashboardStorageKey = "Babylon/InspectorV2/SnippetDashboard/ParticleSystems";
+
+function TryParseJsonString(value: unknown): unknown {
+    if (typeof value !== "string") {
+        return undefined;
+    }
+
+    try {
+        return JSON.parse(value);
+    } catch {
+        return undefined;
+    }
+}
+
+function ParseJsonFileContents(contents: unknown): unknown {
+    if (contents instanceof ArrayBuffer) {
+        const decoder = new TextDecoder("utf-8");
+        return TryParseJsonString(decoder.decode(contents)) ?? undefined;
+    }
+
+    if (typeof contents === "string") {
+        return TryParseJsonString(contents) ?? undefined;
+    }
+
+    return undefined;
+}
+
+function NormalizeParticleSystemSerialization(raw: any): any {

```

Comment:
Can we avoid using `any`? Seems like there are just a few things we expect this to be?